tests/tt_eager/ops/test_sliding_window_ops.cpp

```diff
@@ -0,0 +1,489 @@
+#include "tt_dnn/op_library//sliding_window_op_infra/sliding_window.hpp"
+
+#include "tensor/tensor.hpp"
+
+#include <cstdint>
+#include <cstdlib>
+#include <numeric>
+#include <tuple>
+
+using tt::tt_metal::Tensor;
+using tt::tt_metal::Shape;
+
+
+vector<float> create_filter_vector(owned_buffer::Buffer<bfloat16> &filter_tensor_buf, uint32_t filter_h, uint32_t filter_w) {
+    vector<float> filter_vector;
+    for(auto h = 0; h < filter_h; h++) {
+        for(auto w = 0; w < filter_w; w++) {
+            filter_vector.push_back(filter_tensor_buf[ h * filter_w + w].to_float());
+        }
+    }
+    return filter_vector;
+}
+
+owned_buffer::Buffer<bfloat16> ref_conv_op(
+    Tensor &input_padded_tensor,
+    Shape input_nchw_shape,
+    uint32_t stride_h,
+    uint32_t stride_w,
+    vector<float> &filter_vector,
+    Shape& filter_pyt_tensor_shape,
+    Shape &out_golden_pyt_tensor_shape
+) {
+    auto input_padded_tensor_buf = owned_buffer::get_as<bfloat16>(input_padded_tensor);
+    uint32_t output_n, output_h, output_w;
+    std::tie(output_n, output_h, output_w) = std::tie(out_golden_pyt_tensor_shape[0],
+                                                      out_golden_pyt_tensor_shape[1],
+                                                      out_golden_pyt_tensor_shape[2]);
+    uint32_t filter_h, filter_w;
+    std::tie(filter_h, filter_w) = std::tie(filter_pyt_tensor_shape[0],
+                                            filter_pyt_tensor_shape[1]);
+    uint32_t input_n, input_h, input_w;
+    std::tie(input_n, input_h, input_w) = std::tie(input_nchw_shape[0],
+                                                   input_nchw_shape[1],
+                                                   input_nchw_shape[2]);
+
+    auto out_golden_pyt_tensor = owned_buffer::create<bfloat16>(output_n * output_h * output_w);
+
+    auto out_idx = 0;
+    vector<float> input_window;
+    for(int i = 0; i < output_n; i++) {
+        for (int k = 0; k < output_h; k++) {
+            for(int l = 0; l < output_w; l++) {
+                /* Get input vector of filter size to calculate convolution.*/
+                for(int m = 0; m < filter_h; m++) {
+                    for(int n = 0; n < filter_w; n++) {
+                        auto anchor = i * (input_h * input_w) + k * stride_h * input_w + l * stride_w;
+                        auto idx = anchor + m * input_w * stride_h + n * stride_w;
+                        input_window.push_back(input_padded_tensor_buf[idx].to_float());
+                    }
+                }
+                out_golden_pyt_tensor[out_idx] = bfloat16(static_cast<float>(inner_product(input_window.begin(), input_window.end(), filter_vector.begin(), 0.0)));
+                out_idx++;
+                input_window.clear();
+            }
+        }
+    }
+
+    return out_golden_pyt_tensor;
+}
+
+uint32_t compare_out_with_golden(owned_buffer::Buffer<bfloat16> &out_golden_tensor_buf, owned_buffer::Buffer<bfloat16> &conv_tensor_buf) {
```

Comment:
for comparing the tensors, we can directly use the existing comparison utility functions instead.

Reply:
do you mean == operator?? That will not print error. if that is ok I can use that operator.